src/client/mod.rs

```diff
@@ -287,6 +292,58 @@ impl<'a> ClientBuilder<'a> {
 
         self
     }
+
+    /// Sets how long the http client should wait for the entire response from discord API.
+    /// By default, an http request will never timeout, it could create a deadlock sometimes.
```

Comment:
We prefer British English spelling and pronunciation, where the 'h' in words such as "http" is pronounced rather than being a silent letter. As a result, this affects the choice of the a/an determiner.
```suggestion
    /// By default, a http request will never timeout, it could create a deadlock sometimes.
```